apps/ui/src/pages/applications/sections/sign.tsx

```diff
@@ -0,0 +1,115 @@
+import { type SubmitHandler, useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { useOutletContext } from 'react-router';
+
+import SectionWrapper from '@/components/layouts/SectionWrapper';
+import ESignature from '@/components/pages/application/form-components/ESignature';
+import SectionContent from '@/components/pages/application/SectionContent';
+import SectionFooter from '@/components/pages/application/SectionFooter';
+import SectionTitle from '@/components/pages/application/SectionTitle';
+import { type ApplicationOutletContext } from '@/global/types';
+
+const { Text } = Typography;
+
+const SignAndSubmit = () => {
+	const { t: translate } = useTranslation();
+	const { isEditMode, appId } = useOutletContext<ApplicationOutletContext>();
+	const [openModal, setOpenModal] = useState(false);
+	const [validatedData, setValidatedData] = useState<eSignatureSchemaType | undefined>(undefined);
+	const signatureRef = useRef(null);
+
+	const { handleSubmit, control, setValue, formState, watch, clearErrors, reset } = useForm<eSignatureSchemaType>({
+		resolver: zodResolver(esignatureSchema),
+	});
+
+	const onSubmit: SubmitHandler<eSignatureSchemaType> = (data) => {
+		setOpenModal(true);
+		setValidatedData(data);
+	};
+
+	const modalSubmission = () => {
+		console.log('Submit Clicked!');
+		console.log(validatedData);
+	};
+
+	const watchSignature = watch('signature');
+	const watchCreatedAt = watch('createdAt');
+
+	return (
+		<>
+			<SectionWrapper>
+				<Form layout="vertical" onFinish={handleSubmit(onSubmit)}>
+					<SectionTitle title={translate('sign-and-submit-section.title')} showDivider={false} />
+					<SectionContent
+						showDivider={false}
+						title={translate('sign-and-submit-section.section1.title')}
+						text={translate('sign-and-submit-section.section1.description')}
+					>
+						<Row>
+							<Col xs={{ flex: '100%' }} md={{ flex: '100%' }} lg={{ flex: '100%' }}>
+								<input disabled type="hidden" name="createdAt" />
+								<ESignature
+									signatureRef={signatureRef}
+									name="signature"
+									control={control}
+									watch={watch}
+									formState={formState}
+									setValue={setValue}
+									reset={reset}
+									clearErrors={clearErrors}
+									disableSaveButton={!watchSignature || !!(watchSignature && watchCreatedAt)}
+									disablePreviewButton={!watchSignature}
+									downloadButton={translate('sign-and-submit-section.section1.buttons.download')}
+									saveButton={translate('sign-and-submit-section.section1.buttons.save')}
+									clearButton={translate('sign-and-submit-section.section1.buttons.clear')}
+									previewButton={translate('sign-and-submit-section.section1.buttons.view')}
+								/>
+							</Col>
+						</Row>
+						<Row style={{ minHeight: '40vh' }} />
+					</SectionContent>
+					<SectionFooter currentRoute="sign" isEditMode={isEditMode} onSubmit={handleSubmit(onSubmit)} />
```

Comment:
`SectionFooter` should be updated so that it uses the passed `onSubmit` prop and the modal can be tested
Currently `SectionFooter` is `disabled` and using placeholder logic instead of the `onSubmit` prop